### PR TITLE
Add VCS-based sidebar label coloring

### LINKED ISSUE #188: Open files close icon margin issue after setting 'spacegray_fileicons:true'
After setting 'spacegray_fileicons:true', the open files close icon is not totally shown, only part of it.  Resizing window doesn't work. 

spacegray-open-file-icon-issue

### LINKED ISSUE #214: Git + sidebar + gutter
Sublime Text now has a simple Git integration. Problem is, the dots it displays in the sidebar are distracting for those who are used to depending on the gutter indicators (e.g. from GitGutter or SublimeLinter). This is being discussed in [this forum post](https://forum.sublimetext.com/t/git-file-status-badges/40694).

Currently there's no alternative display mode, so the only fix, besides outright disabling the integration, is theming. Thankfully, as mentioned in that post, the necessary adjustments have already been made in another theme - [Toxin](https://github.com/p3lim/sublime-toxin) - so this would be relatively easy to replicate in Spacegray.

However, would such a change be welcome? @marcofugaro @kkga

### PR DESCRIPTION
**Pull request by alecmev**
Essentially a copy-paste from [Toxin](https://github.com/p3lim/sublime-toxin), thanks @p3lim! One divination is that I chose to decolor the VCS badges, instead of outright hiding them, since they help distinguish created/modified from staged. Another is the italic for ignored files, which I wouldn't mind dropping, but it's somewhat useful.

Didn't know what colors to choose, so just looked at [VS Code's Base16 Themes](https://marketplace.visualstudio.com/items?itemName=AndrsDC.base16-themes).

This is opt-in, `spacegray_sidebar_vcs_label` setting enables the whole thing, `spacegray_sidebar_vcs_expanded` also applies the coloring to expanded folders. ~The latter one is the same setting name as in Toxin, for potential interop.~

Added `variables` introduced in 3179, because why not. Opted not to indent the rules as to not ruin the diff for the main feature and to not pollute the blame trail. `base` prefix because [reasons](https://www.w3.org/TR/CSS22/syndata.html#value-def-identifier).

Also piggybacked in a fix for #188, since the original PR has died a slow and painful death.

Fixes #214
Fixes #188

image

### CODE REVIEW

**Comment by marcofugaro** [comment]
Thanks for the contribution! Will test this locally and review the changes!

**Comment by jeremejevs** [comment]
@marcofugaro Took a closer look at the other themes, besides Eighties, and yeah, those needed tuning, especially Light. Also got rid of the selected effect, doesn't look good, and isn't really necessary.

**Comment by jeremejevs** [comment]
Rebased to fix a conflict in readme.

**Comment by alecmev** [comment]
@marcofugaro

**Comment by alecmev** [comment]
@kkga @marcofugaro Hi! I'm willing to adopt this theme. Could fork, but I don't intend to change much, just modernize.

**Comment by kkga** [comment]
@alecmev hey! Thanks for your interest. Do you want to move this repo to community packages org?

**Comment by alecmev** [comment]
@kkga Yeah! I haven't coordinated with them yet though. I think you need to be a part of an organization to be able to transfer to it. You can transfer to me, and I can take of the rest, what do you prefer?

**Comment by kkga** [comment]
@alecmev any idea what this means?

image

**Comment by FichteFoll** [comment]
Ah, I suspect that is because of an already existing fork. I didn't think of that.

You may also transfer the repo to me and I will take care of it.

**Comment by alecmev** [comment]
> Ah, I suspect that is because of an already existing fork. I didn't think of that.

Indeed! I renamed the fork to `spacegray-fork` ahead of it, but that didn't help.

> You may also transfer the repo to me and I will take care of transfering it to the org.

I already submitted a "detach a fork" request to GitHub, but this is faster, thanks for offering! @kkga Can you transfer to @FichteFoll instead, please?

**Comment by alecmev** [comment]
@kkga The fork has been detached, just in case, so you can transfer to me too :wink:

**Comment by FichteFoll** [comment]
Indeed, all done on my part.

**Comment by kkga** [comment]
Great, thanks all!

**Comment by alecmev** [comment]
Thank you! :+1:

**Comment by jakitliang** [comment]
Is there any progress?

**Comment by alecmev** [comment]
Didn't realize there was demand, will try to clean up and release soon. This isn't ready as-is yet, unfortunately, as I've noticed some inconsistencies between dark/light in personal use.

File: Spacegray.sublime-theme [hunk 1]
```diff
@@ -544,11 +565,190 @@
...
+//
+// SIDEBAR - VCS
+//
+
+    // Decolor default VCS badges
+
+    {
+        "class": "vcs_status_badge",
+        "settings": ["vcs_color"],
+        "layer0.tint": "var(base02)"
```
Comment (FichteFoll) [inline Spacegray.sublime-theme:578]: Is the point of the setting that it only colors the text and not the icon? Seems a bit weird to be, but since it's optional I don't really mind.

However, that also means that there's no option currently to enable both, text and icon coloring, right?

**Review by FichteFoll — COMMENTED** [review]
I took a brief look at the contents here (as a result of #221) and have a small question.

File: Spacegray.sublime-theme (hunk 1 above)
Comment (alecmev) [reply Spacegray.sublime-theme:578]: Thank you for taking a look!

I think I did this as to not have anything colorful next to the gutter, otherwise it can get noisy, and gutter's signals are more important. The icons aren't decolored when the VCS integration isn't on because otherwise it becomes hard to parse the sidebar for VCS info.

Toxin, the main inspiration, hid the icons completely, for example.

> However, that also means that there's no option currently to enable both, text and icon coloring, right?

Correct. Can be added later if somebody wants it, what do you think?

Comment (FichteFoll) [reply Spacegray.sublime-theme:578]: Indeed, this is not a blocker, although when making this configurable separately I would probably choose different setting names to more clearly indicate which part of the siedebar they are controlling.

Comment (alecmev) [reply Spacegray.sublime-theme:578]: Ah, setting names were adapted from Toxin, but it can be changed, since that theme is all but obsolete. What do you propose?

Comment (FichteFoll) [reply Spacegray.sublime-theme:578]: I'm thinking in the lines of "vcs_sidebar_label", leaving the option open for another "vcs_sidebar_icon" setting.

Comment (alecmev) [reply Spacegray.sublime-theme:578]: Something odd is going on with GitHub (contacted support about it), the branch isn't updating, but in the meanwhile, check [this diff](https://github.com/alecmev/spacegray/commit/8fa0a05c808e03bc4fcab0b7488c6ff1291a25d4), please, and let me know if it looks good to you.

Comment (FichteFoll) [reply Spacegray.sublime-theme:578]: That looks perfect to me, thanks for making the adjustments. Feel free to ping me when the change arrived at this PR or alternatively just open a new one.

Comment (alecmev) [reply Spacegray.sublime-theme:578]: Ping! 😉

**Review by FichteFoll — APPROVED** [review]

**Comment by FichteFoll** [comment]
Thanks for sticking to it. I have released [1.4.0](https://github.com/SublimeText/Spacegray/releases/tag/1.4.0) with these changes.

I'm also adding you as a collaborator since you obviously still care about this theme and have shown interest in improving it (so I don't have to 😉 ).

**Comment by alecmev** [comment]
Thanks for taking care of this!
